src/tests/test_log_client.py

```diff
@@ -0,0 +1,226 @@
+            self.log_client.connect()
+            self.assertTrue(self.log_client.connected)
+
+    def test_pull_message(self):
+        self.assertEqual(None, self.log_client.pull_message())
+
+    def test_write_to_local_log(self):
+        self.log_client.write_to_local_log({'message': 'this is a message'})
+        local_log_file = self.log_client.dump_file
+        self.assertTrue(os.path.exists(local_log_file))
+        os.remove(local_log_file)
+
+
+class TestMisc(unittest.TestCase):
+    
+    def setUp(self):
+        self.user = pwd.getpwuid(os.getuid())[0]
+
+    def test_genMsgType(self):
+        """
+        test the gentMsgType function which formats
+        the message type for the log server
+
+        the list test_lines was derived from the log_client.py file with the
+        following command:
+        grep 'if string.find.*lowLine' ../log_client.py | sed -e 's/^.*Line, //g' -e 's/\\(".*"\\)\\(.*\\)/\1/'| sort | tr "\n" ","
+        sometimes members in test_lines are repeated, the same input line
+        is handled by different if statements
+        """
+        test_lines = [
+            " mover ", " vol", "(re)", "added to mover list", "log server", "log_server",
+                       "backup", "bad", "badmount", "busy_vols", "busy_vols", "cantrestart",
+                       "completed precautionary", "config server", "configuration server",
+                       "dismount", "done", "encp", "exception", "file clerk", "file database",
+                       "file", "file_clerk ", "find_mover", "find_mover", "full", "get_suspect_vol",
+                       "get_suspect_vol", "get_user_socket", "get_user_socket", "get_work",
+                       "get_work", "get_work", "get_work", "getmoverlist", "getmoverlist",
+                       "getwork", "getwork", "hurrah", "insert", "insertvol", "library manager",
+                       "library_manager ", "load", "load", "media changer", "media_changer ",
+                       "next_work", "next_work", "no such vol", "open_file_write",
+                       "open_file_write", "performing precautionary", "quit", "read ",
+                       "reading", "restart", "root error", "root_error ", "serverdied",
+                       "start", "start{", "stop", "tape stall", "tape_stall", "unbind vol",
+                       "unbind", "unload", "unmount", "update_mover_list", "update_mover_list",
+                       "volume clerk", "volume database", "volume_clerk ", "wrapper.write",
+                       "wrapper.write", "write ", "write_to_hsm", "writing"
+        ]
+
+        for lowline in test_lines:
+            for err in e_errors.stypedict:
+                gen_msg = log_client.genMsgType(lowline, '', err)
+                #print "debug gen_msg=%s " % gen_msg
+                self.assertNotEqual(
+                    "", log_client.genMsgType(
+                        lowline, lowline, err))
```

Comment:
is the first arg here supposed to be gen_msg?

Reply:
I am not 100% sure as this fairly complicated method is not used anywhere.  I just tested that it spat something out for all legal inputs

Reply:
Ack, just checking as you don't actually seem to use gen_msg set on 219 unless I'm missing something!